[AIRFLOW-40] add DAGs' filtering over ldap groups

dsjl opened the pull request:
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- https://issues.apache.org/jira/browse/AIRFLOW-40

bolkedebruin commented on airflow/www/views.py:
```diff
@@ -1725,14 +1725,38 @@ def index(self):
+
+        if do_filter and owner_mode not in ['user', 'ldapgroup']:
+            logout_user()
+            flash("Configuration error: "
```
I think it is better to move this to configuration._validate

dsjl replied:
Would adding an 'elif' in ConfigParserWithDefaults._validate be sufficient or do you know if a better config validation will happen in the near future?

bolkedebruin commented on airflow/www/views.py:
```diff
@@ -1725,14 +1725,25 @@ def index(self):
-                    DM.owners == current_user.username)
-                    .all()
-            )
+            if owner_mode == 'ldapgroup':
```
This can be refactored to be a bit cleaner and not repeat code too much. qry = qry.filter()

dsjl committed:
[AIRFLOW-40] Adding LDAP group filtering feature.

It is now possible to filter over LDAP group (in the web interface) when using the LDAP authentication backend.
Note that this feature requires the "memberOf" overlay to be configured on the LDAP server.
